### PR TITLE
Allow for named fonts

### PR DESCRIPTION
**Pull request by HenryTheCat**
Allow naming of fonts, as in the `runic` example in the [bear_lib_terminal doc](http://foo.wyrd.name/en:bearlibterminal:reference:configuration#font_and_tileset_management).

### CODE REVIEW

**Comment by HenryTheCat** [comment]
@nabijaczleweli I opened the PR without opening an issue first because it looked like a minimal, simple addition. Let me know if firther work is needed on this.

**Comment by HenryTheCat** [comment]
The only reason was lazyness, but I tested it now.

It seems to work alright, at least as far as configuration goes, but then I have issues related to codepage. I used `Origin::Named` to load a tileset (a PNG with 16x16 tiles of 16x16 pixels each); since no offset is specified, its codepage starts at `0x0000`. Though, when I try to print, `\u{0000}` causes a panic, `\u{0001}` to `\u{0009}` work as expected, `\u{000A}` selects a blank tile. The `\u{001X}` series seems to work fine, etc…

I don't know what the issue is, but I suspect that those codes are reserved for something special. On the original BearLibTerminal documentation I found no such example, as bitmap fonts are always loaded with a codepage and tiles with an offset.

Long story short, I think the patch works as intended, and that the issue is due to my incorrect usage of codepages. It isn't clear from the BearLibTerminal documentation whether it's possible to set a font with both a name and an offset (which would not be allowed by the current patch), but I don't know if that really makes much sense anyway.

**Comment by nabijaczleweli** [comment]
Ah, it's good to hear it works.

Could you, however, change the type from `String` to `Cow<'static, str>`? Will help avoid some allocations.

**Comment by HenryTheCat** [comment]
I didn't know about `Cow`s (thanks for suggesting that). I think I fixed it as you intended.

**Comment by nabijaczleweli** [comment]
Ah, lovely! Rebase and this is 👌 to go

**Comment by HenryTheCat** [comment]
Sorry I'm not sure what I should. My git is really poor!

**Comment by nabijaczleweli** [comment]
You should be able to just
```
git fetch https://github.com/nabijaczleweli/BearLibTerminal.rs
git rebase FETCH_HEAD
git push -f
```

On the `patch-2` branch

**Comment by nabijaczleweli** [comment]
Sorry for the delay, I missed the notification when you force-pushed originally!

**Comment by nabijaczleweli** [comment]
Released in [`v2.0.0`](https://github.com/nabijaczleweli/BearLibTerminal.rs/releases/tag/v2.0.0).
